tools/src/main/scala/org/apache/spark/tools/GenerateMIMAIgnore.scala

```diff
@@ -105,16 +109,19 @@ object GenerateMIMAIgnore {
 
   /** Scala reflection does not let us see inner function even if they are upgraded
     * to public for some reason. So had to resort to java reflection to get all inner
-    * functions with $$ in there name.
+    * functions with $$ in their name.
     */
-  def getInnerFunctions(classSymbol: unv.ClassSymbol): Seq[String] = {
+  def getInnerFunctions(className: String): Seq[String] = {
     try {
-      Class.forName(classSymbol.fullName, false, classLoader).getMethods.map(_.getName)
-        .filter(_.contains("$$")).map(classSymbol.fullName + "." + _)
```

Comment:
In addition to the reformatting here, I also changed `classSymbol.fullName` to `className`. The problem is that `classSymbol.fullName` uses underscores instead of dollar signs when denoting inner classes and had some other weirdness, so the old name was causing `ClassNotFoundException` when it was passed to `Class.forName`. This caused the flood of `"Unable to detect inner functions" warnings in the MiMa logs, e.g.

```
[WARN] Unable to detect inner functions for class:org.apache.spark.sql.catalyst.parser.SparkSqlParser_FromClauseParser.partitionTableFunctionSource_return
[WARN] Unable to detect inner functions for class:org.apache.spark.sql.api.r.SQLUtils.RegexContext
[WARN] Unable to detect inner functions for class:org.apache.spark.sql.catalyst.parser.SparkSqlParser_ExpressionParser.$DFA21
```